### ISSUE TITLE
Extend DrawParameters to control GL_CLIP_PLANEi

### ISSUE THREAD

**Issue by mkovacs** [glium/glium#1662]
OpenGL offers the functionality to clip primitives after the vertex stage against arbitrary planes.
Glium should expose controls for turning the clip planes on and off.

Links:
https://www.khronos.org/registry/OpenGL-Refpages/gl4/html/gl_ClipDistance.xhtml
https://www.khronos.org/registry/OpenGL-Refpages/gl2.1/xhtml/glClipPlane.xml

**Comment by tomaka** [comment]
I'm not sure ; isn't that purely a shader-controlled thing?

**Comment by mkovacs** [comment]
No, the clip planes still need to be enabled, e.g. `gl.Enable(gl::CLIP_DISTANCE0);` or  `gl.Enable(gl::CLIP_PLANE0);`, otherwise writing to `gl_ClipDistance[0]` is ineffective (I tested this).
